### ISSUE TITLE
Dependency Dashboard stopped working

### ISSUE THREAD

**Issue by enricojonas** [whitesource/renovate-on-prem#176]
Hi,

For some reason our dependency dashboards do not work anymore. Nothing happens if you check any dependency or the manual run checkbox. Following entries are in the log:

```
2022-01-07 12:25:11.489 CET
renovate-container
DEBUG: Received webhook event (repository=xxx/xxx)
Info
2022-01-07 12:25:11.489 CET
renovate-container
"event": "issues",
Info
2022-01-07 12:25:11.489 CET
renovate-container
"action": "edited",
Info
2022-01-07 12:25:11.489 CET
renovate-container
"installationId": 696
Info
2022-01-07 12:25:11.489 CET
renovate-container
DEBUG: Ignoring issue action: edited (repository=xxx/xxx)
```

We did not change any configuration recently.

Before, when editing the dashboard, following entries where visible. DEBUG was not enabled so there is no more info for old entries:

```
2021-12-22 15:23:37.706 CET
renovate-container
INFO: Event result
Info
2021-12-22 15:23:37.706 CET
renovate-container
"result": {
Info
2021-12-22 15:23:37.706 CET
renovate-container
"task": "enqueue",
Info
2021-12-22 15:23:37.706 CET
renovate-container
"installationId": 696,
Info
2021-12-22 15:23:37.707 CET
renovate-container
"repository": "xxx/xxx",
Info
2021-12-22 15:23:37.707 CET
renovate-container
"reason": "master-issue-check"
Info
2021-12-22 15:23:37.707 CET
renovate-container
}
Info
2021-12-22 15:23:37.955 CET
renovate-container
INFO: Processing repository (repository=xxx/xxx)
Info
2021-12-22 15:23:37.955 CET
renovate-container
"priority": 17,
Info
2021-12-22 15:23:37.956 CET
renovate-container
"reason": "master-issue-check"
```

Versions:

Renovate-On-Prem 1.6.2
GitHub Enterprise Server 3.2.6 (This was updated recently, not sure if it causes any issue)

This is the config.js (except hostrules and packagerules:

```
    "autodiscover": false,
    "requireConfig": false,
    "branchPrefix": "renovate-",
    "dependencyDashboard": true,
    "dependencyDashboardAutoclose": true,
    "dependencyDashboardFooter": "\n- [ ] <!-- manual job -->Check this box to trigger a request for Renovate to run again on this repository\n",
    "prCreation": "approval",
```

**Comment by rarkins** [comment]
What do the logs say about the dashboard issue when Renovate runs against the same repo?

**Comment by enricojonas** [comment]
@rarkins Always the same log entries. It's not possible to trigger anything through the dashboard. I closed the dashboard manually and renamed the onboarding PR. New onboarding PR was created. After closing the onboarding PR, a new Dashboard was created. It acts in the same way though - nothing happens when ticking any checkbox and above log entries in the log. Also re-started renovate while enabling debug - same issue.

**Comment by enricojonas** [comment]
@rarkins

Here some logfiles, I offboarded the repo and onboarded again, closed the onboarding PR and the dashboard got created. Then checked the first issue on the dashboard.

[onboard.txt](https://github.com/whitesource/renovate-on-prem/files/7829392/onboard.txt)
[close_onboarding_pr.txt](https://github.com/whitesource/renovate-on-prem/files/7829390/close_onboarding_pr.txt)
[check_issue_in_dashboard.txt](https://github.com/whitesource/renovate-on-prem/files/7829402/check_issue_in_dashboard.txt)

![image](https://user-images.githubusercontent.com/46678516/148559828-1343b6f6-c066-45f7-8081-02449acd77a4.png)

**Comment by rarkins** [comment]
Can you extract the debug logs for the branch in question plus any other mentions of "dashboard", from the last log?

**Comment by enricojonas** [comment]
Honestly, I think it's a github bug. I just checked the webhooks when something is checked in the issue:

```
    "body": "This issue provides visibility into Renovate updates and their statuses. [Learn more](https://docs.renovatebot.com/key-concepts/dashboard/)\n\n## Pending Status Checks\n\nThese updates await pending status checks. To force their creation now, click the checkbox below.\n\n - [x] <!-- approvePr-branch=renovate-all-patch -->Update dependency org.apache.maven.plugins:maven-release-plugin to v3.0.0-M5\n - [ ] <!-- approvePr-branch=renovate-all-minor -->Update all dependencies - minor (`org.sonarsource.scanner.maven:sonar-maven-plugin`, `com.fasterxml.jackson.core:jackson-core`)\n - [ ] <!-- approvePr-branch=renovate-all-spring-minor -->Update dependency org.springframework.boot:spring-boot-starter-parent to v2.6.2\n\n---\n\n- [x] <!-- manual job -->Check this box to trigger a request for Renovate to run again on this repository\n\n",
    "performed_via_github_app": null
  },
  "changes": {

  },
```

As you can see the changes are empty. In an old webhook (when it was working) the changes were there. I guess renovate will only do something when there are changes.

This is from an old webhook:

```
    "body": "This issue provides visibility into Renovate updates and their statuses. [Learn more](https://docs.renovatebot.com/key-concepts/dashboard/)\n\n## PR Creation Approval Required\n\nThese branches exist but PRs won't be created until you approve them by clicking on a checkbox.\n\n - [ ] <!-- approvePr-branch=renovate-all-terraform-minor-patch -->Update all non-major terraform dependencies (`google`, `hashicorp/terraform`)\n - [ ] <!-- approvePr-branch=renovate-major-all-terraform-major -->Update Terraform google to v4\n - [ ] <!-- approvePr-branch=renovate-google-cloud-functions-framework-2.x -->Update dependency @google-cloud/functions-framework to v2\n - [x] <!-- approvePr-branch=renovate-eslint-8.x -->Update dependency eslint to v8\n - [ ] <!-- approvePr-branch=renovate-inversify-6.x -->Update dependency inversify to v6\n - [ ] <!-- approvePr-branch=renovate-sinon-12.x -->Update dependency sinon to v12\n\n## Ignored or Blocked\n\nThese are blocked by an existing closed PR and will not be recreated unless you click a checkbox below.\n\n - [ ] <!-- recreate-branch=renovate-jest-27.x -->[Update dependency @types/jest to v27](../pull/16)\n - [ ] <!-- recreate-branch=renovate-typescript-eslint-eslint-plugin-5.x -->[Update dependency @typescript-eslint/eslint-plugin to v5](../pull/17)\n - [ ] <!-- recreate-branch=renovate-typescript-eslint-parser-5.x -->[Update dependency @typescript-eslint/parser to v5](../pull/19)\n - [ ] <!-- recreate-branch=renovate-audit-ci-5.x -->[Update dependency audit-ci to v5](../pull/20)\n\n---\n\n- [x] <!-- manual job -->Check this box to trigger a request for Renovate to run again on this repository\n\n",
    "performed_via_github_app": null
  },
  "changes": {
    "body": {
      "from": "This issue provides visibility into Renovate updates and their statuses. [Learn more](https://docs.renovatebot.com/key-concepts/dashboard/)\n\n## PR Creation Approval Required\n\nThese branches exist but PRs won't be created until you approve them by clicking on a checkbox.\n\n - [ ] <!-- approvePr-branch=renovate-all-terraform-minor-patch -->Update all non-major terraform dependencies (`google`, `hashicorp/terraform`)\n - [ ] <!-- approvePr-branch=renovate-major-all-terraform-major -->Update Terraform google to v4\n - [ ] <!-- approvePr-branch=renovate-google-cloud-functions-framework-2.x -->Update dependency @google-cloud/functions-framework to v2\n - [x] <!-- approvePr-branch=renovate-eslint-8.x -->Update dependency eslint to v8\n - [ ] <!-- approvePr-branch=renovate-inversify-6.x -->Update dependency inversify to v6\n - [ ] <!-- approvePr-branch=renovate-sinon-12.x -->Update dependency sinon to v12\n\n## Ignored or Blocked\n\nThese are blocked by an existing closed PR and will not be recreated unless you click a checkbox below.\n\n - [ ] <!-- recreate-branch=renovate-jest-27.x -->[Update dependency @types/jest to v27](../pull/16)\n - [ ] <!-- recreate-branch=renovate-typescript-eslint-eslint-plugin-5.x -->[Update dependency @typescript-eslint/eslint-plugin to v5](../pull/17)\n - [ ] <!-- recreate-branch=renovate-typescript-eslint-parser-5.x -->[Update dependency @typescript-eslint/parser to v5](../pull/19)\n - [ ] <!-- recreate-branch=renovate-audit-ci-5.x -->[Update dependency audit-ci to v5](../pull/20)\n\n---\n\n- [ ] <!-- manual job -->Check this box to trigger a request for Renovate to run again on this repository\n\n"
    }
```


The branch where the PR should have been created for is "renovate-all-patch".

```
               {
                 "datasource": "maven",
                 "depName": "org.apache.maven.plugins:maven-release-plugin",
                 "currentValue": "3.0.0-M4",
                 "fileReplacePosition": 2073,
                 "registryUrls": ["https://repo.maven.apache.org/maven2"],
                 "depIndex": 4,
                 "updates": [
                   {
                     "bucket": "non-major",
                     "newVersion": "3.0.0-M5",
                     "newValue": "3.0.0-M5",
                     "releaseTimestamp": "2022-01-02T10:21:45.000Z",
                     "newMajor": 3,
                     "newMinor": 0,
                     "updateType": "patch",
                     "branchName": "renovate-all-patch"
                   }
                 ],
                 "warnings": [],
                 "versioning": "maven",
                 "sourceUrl": "http://svn.apache.org/viewvc/maven/release/tags/maven-release-2.5.3",
                 "homepage": "http://maven.apache.org/",
                 "currentVersion": "3.0.0-M4",
                 "isSingleVersion": true,
                 "fixedVersion": "3.0.0-M4"
               }
             ],
             "parent": "../pom.xml"
           }
         ]
       }
DEBUG: processRepo() (repository=ourorg/ourrepo)
DEBUG: Processing 3 branches: renovate-all-minor, renovate-all-patch, renovate-all-spring-minor (repository=ourorg/ourrepo)
DEBUG: Calculated maximum PRs remaining this run (repository=ourorg/ourrepo)
       "prsRemaining": 99
DEBUG: PullRequests limit = 99 (repository=ourorg/ourrepo)
DEBUG: Calculated maximum branches remaining this run (repository=ourorg/ourrepo)
       "branchesRemaining": 99
DEBUG: Branches limit = 99 (repository=ourorg/ourrepo)
DEBUG: Setting current branch to master (repository=ourorg/ourrepo, branch=renovate-all-patch)
DEBUG: latest commit (repository=ourorg/ourrepo, branch=renovate-all-patch)
       "branchName": "master",
       "latestCommitDate": "2021-11-25T14:55:20+00:00"
DEBUG: getBranchPr(renovate-all-patch) (repository=ourorg/ourrepo, branch=renovate-all-patch)
DEBUG: findPr(renovate-all-patch, undefined, open) (repository=ourorg/ourrepo, branch=renovate-all-patch)
DEBUG: findPr(renovate-all-patch, undefined, closed) (repository=ourorg/ourrepo, branch=renovate-all-patch)
DEBUG: branchExists=false (repository=ourorg/ourrepo, branch=renovate-all-patch)
DEBUG: dependencyDashboardCheck=undefined (repository=ourorg/ourrepo, branch=renovate-all-patch)
DEBUG: recreateClosed is false (repository=ourorg/ourrepo, branch=renovate-all-patch)
DEBUG: findPr(renovate-all-patch, Update dependency org.apache.maven.plugins:maven-release-plugin to v3.0.0-M5, !open) (repository=ourorg/ourrepo, branch=renovate-all-patch)
DEBUG: prAlreadyExisted=false (repository=ourorg/ourrepo, branch=renovate-all-patch)
DEBUG: Checking schedule(at any time, null) (repository=ourorg/ourrepo, branch=renovate-all-patch)
DEBUG: No schedule defined (repository=ourorg/ourrepo, branch=renovate-all-patch)
DEBUG: Branch needs creating (repository=ourorg/ourrepo, branch=renovate-all-patch)
DEBUG: Using reuseExistingBranch: false (repository=ourorg/ourrepo, branch=renovate-all-patch)
DEBUG: manager.getUpdatedPackageFiles() reuseExistinbranch=false (repository=ourorg/ourrepo, branch=renovate-all-patch)
DEBUG: Updating org.apache.maven.plugins:maven-release-plugin in pom.xml (repository=ourorg/ourrepo, branch=renovate-all-patch)
DEBUG: Updated 1 package files (repository=ourorg/ourrepo, branch=renovate-all-patch)
DEBUG: No updated lock files in branch (repository=ourorg/ourrepo, branch=renovate-all-patch)
DEBUG: 1 file(s) to commit (repository=ourorg/ourrepo, branch=renovate-all-patch)
DEBUG: Committing files to branch renovate-all-patch (repository=ourorg/ourrepo, branch=renovate-all-patch)
DEBUG: Setting git author name (repository=ourorg/ourrepo, branch=renovate-all-patch)
       "gitAuthorName": "renovate[bot]"
DEBUG: Setting git author email (repository=ourorg/ourrepo, branch=renovate-all-patch)
       "gitAuthorEmail": "renovate[bot]@users.noreply.git.server.com"
DEBUG: git commit (repository=ourorg/ourrepo, branch=renovate-all-patch)
       "deletedFiles": [],
       "ignoredFiles": [],
       "result": {
         "author": null,
         "branch": "renovate-all-patch",
         "commit": "82df9a8",
         "root": false,
         "summary": {"changes": 1, "insertions": 1, "deletions": 1}
       }
DEBUG: git push (repository=ourorg/ourrepo, branch=renovate-all-patch)
       "result": {
         "pushed": [
           {
             "deleted": false,
             "tag": false,
             "branch": true,
             "new": true,
             "alreadyUpdated": false,
             "local": "refs/heads/renovate-all-patch",
             "remote": "refs/heads/renovate-all-patch"
           }
         ],
         "branch": {
           "local": "renovate-all-patch",
           "remote": "renovate-all-patch",
           "remoteName": "origin"
         },
         "ref": {"local": "refs/remotes/origin/renovate-all-patch"},
         "remoteMessages": {
           "all": [
             "Create a pull request for 'renovate-all-patch' on GitHub by visiting:",
             "https://git.server.com/ourorg/ourrepo/pull/new/renovate-all-patch"
           ],
           "pullRequestUrl": "https://git.server.com/ourorg/ourrepo/pull/new/renovate-all-patch"
         }
       }
DEBUG: LOG_CONTEXT: undefined (repository=ourorg/ourrepo, branch=renovate-all-patch)
DEBUG: Received webhook event (repository=ourorg/ourrepo, branch=renovate-all-patch)
       "event": "push",
       "action": undefined,
       "installationId": 696
DEBUG: Ignoring non-bot commit (repository=ourorg/ourrepo, branch=renovate-all-patch)
 INFO: Branch created (repository=ourorg/ourrepo, branch=renovate-all-patch)
       "commitSha": "82df9a8"
DEBUG: Branch status pending (repository=ourorg/ourrepo, branch=renovate-all-patch)
       "commitSha": "82df9a8"
```

Those are the mentions of Dashboard in the log when you close the onboarding PR:

```
DEBUG: findIssue(Dependency Dashboard) (repository=ourorg/ourrepo)
DEBUG: dependencyDashboardCheck=undefined (repository=ourorg/ourrepo, branch=renovate-all-patch)
DEBUG: dependencyDashboardCheck=undefined (repository=ourorg/ourrepo, branch=renovate-all-minor)
DEBUG: dependencyDashboardCheck=undefined (repository=ourorg/ourrepo, branch=renovate-all-spring-minor)
DEBUG: Ensuring Dependency Dashboard (repository=ourorg/ourrepo)
DEBUG: ensureIssue(Dependency Dashboard) (repository=ourorg/ourrepo)
```

However, when I check the first checkbox "Update dependency org.apache.maven.plugins:maven-release-plugin to v3.0.0-M5" nothing is happening. Expectation would be that a PR is created against "renovate-all-patch" branch. Those are the only log entries written. We are using the on prem setup with github APP so renovate is triggered by the events / webhooks github is sending. I don't understand why this happens since it was working before as expected and the renovate installation is stateless. It seems that renovate is ignoring any update to the issue

```
DEBUG: LOG_CONTEXT: undefined
DEBUG: Received webhook event (repository=ourorg/ourrepo)
       "event": "issues",
       "action": "edited",
       "installationId": 696
DEBUG: Ignoring issue action: edited (repository=ourorg/ourrepo)
DEBUG: No job - sleeping 5s
DEBUG: No job - sleeping 5s

```

**Comment by rarkins** [comment]
This line:

DEBUG: dependencyDashboardCheck=undefined (repository=ourorg/ourrepo, branch=renovate-all-patch)

Implies that Renovate did not think the dashboard issue was checked for that branch at the time it ran

**Comment by enricojonas** [comment]
Yes, since the dashboard was not yet existing that is correct. It was created by renovate at this time. But for any change in the dashboard after dashboard creation, only following log entries are visible. How can renovate determine changes if the "changes" section is missing in the webhook?!

```
DEBUG: LOG_CONTEXT: undefined
DEBUG: Received webhook event (repository=ourorg/ourrepo)
       "event": "issues",
       "action": "edited",
       "installationId": 696
DEBUG: Ignoring issue action: edited (repository=ourorg/ourrepo)
DEBUG: No job - sleeping 5s
DEBUG: No job - sleeping 5s
```

**Comment by rarkins** [comment]
True. Renovate looks for the new issue body content in the webhook to determine if it includes any checked boxes. If it doesn't find any (or changes is missing as you observed) then it won't trigger anything

**Comment by enricojonas** [comment]
@rarkins github confirmed that it's a bug on their side. We need to upgrade GHES again to fix it. Thank you for looking into it anyway.
